### ISSUE TITLE
eigen value of cov matrix

### ISSUE THREAD

**Issue by hcyoo93** [pmorerio/minimal-entropy-correlation-alignment#4]
Hello, @pmorerio 

I got another question about the eigenvalue of cov_matrix.

As far as I understood from your paper, cov_representation is positive definite, therefore
taking a logarithm on the eigenvalues makes sense.

However, when I decompose my covariance matrices I got some negative eigenvalues which result in `nan` logarithm values.

1. Is this expected behavior?

2. If so, `coral_loss` can perform as well as `log_coral_loss`?

I am afraid that I miss some part of your paper.
Any comments are welcome!!

Best,
Yoo

**Comment by pmorerio** [comment]
Hi,
yes, in principle a covariance matrix should be positive definite.
However it can happen that negative eigenvalues pop out due to numerical approximations.
A you say, this can result in `nan` values.
For this reason it is common practice to regularize the covariance matrix as I commented in https://github.com/pmorerio/minimal-entropy-correlation-alignment/blob/3b261683fbb24eca2d2efb6dff5c25ec4301b7fc/svhn2mnist/model.py#L65
Basically you want to add small values to the diagonal as suggested [here](https://stats.stackexchange.com/questions/390532/adding-a-small-constant-to-the-diagonals-of-a-matrix-to-stabilize).

`coral_loss` is suboptimal, but of course is a viable option if you are stuck with numerical issues.

Best,
P.

**Comment by hcyoo93** [comment]
Thank you for your answer!

I will try the regularization for numerical stability!

Best,
Yoo

**Comment by pmorerio** [comment]
Hope it works,
let me know.

Best,
P.
